conformance/tests/mesh-ports.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/echo"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, MeshPorts)
+}
+
+var MeshPorts = suite.ConformanceTest{
+	ShortName:   "MeshPorts",
+	Description: "A mesh route can optional configure 'port' in parentRef",
+	Features: []suite.SupportedFeature{
+		suite.SupportMesh,
```

Comment:
I think this test is also exercising the extended ResponseHeaderModifier feature:

```suggestion
		suite.SupportMesh,
		suite.SupportHTTPRoute,
		suite.SupportHTTPResponseHeaderModification,
```